ecs-agent/netlib/data/network_client.go

```diff
@@ -94,3 +132,127 @@ func (ndc *networkDataClient) GetNetworkNamespace(netNSName string) (*tasknetwor
 
 	return netNS, nil
 }
+
+// AssignGeneveDstPort returns an unused destination port number for GENEVE interfaces.
+// By default for a particular VNI, it will return the default GENEVE destination port - 6081.
+// In case port 6081 is taken by another interface using the same VNI, it will chose a
+// random port from within the pre-configured range.
+// Once a port number is assigned by this function, it won't be used until
+// ReleaseGeneveDstPort is called with the port and VNI.
+func (ndc *networkDataClient) AssignGeneveDstPort(vni string) (uint16, error) {
+	metricEntry := ndc.metricsFactory.New(metrics.AssignGeneveDstPortMetricName)
+	port, err := ndc.assignPort(geneveDstPortMin, geneveDstPortMax, vni, geneveDstPortDefault)
+	metricEntry.Done(err)
+
+	if errNoValue == errors.Cause(err) {
+		ndc.metricsFactory.New(metrics.V2NDestinationPortExhaustedMetricName).Done(err)
+	}
+	return port, err
+}
+
+func (ndc *networkDataClient) assignPort(min, max int, vni string, defaultValue int) (uint16, error) {
```

Comment:
Should we place `defaultValue` next to min and max so we can declare all 3 of them as int together?

Reply:
I have removed implementation of NetworkDataClient from the shared library because the agent is able to inject the same into the network builder.